File: Entities/Invoice.cs
namespace PrivateHospitalSystem.Entities
{
    public enum InvoiceStatus
    {
        Pending,
        Paid,
        Cancelled
    }

    public class Invoice
    {
        public Guid Id { get; set; }

        public Guid PatientId { get; set; }
        public Patient? Patient { get; set; }

        public string ProcedureType { get; set; } = string.Empty; 
        public decimal TotalAmount { get; set; }
        public decimal InsuranceCoveredAmount { get; set; }
        public decimal PatientAmount { get; set; } 

        public InvoiceStatus Status { get; set; } = InvoiceStatus.Pending;

        public DateTime IssuedAt { get; set; } = DateTime.UtcNow;
        public DateTime? PaidAt { get; set; }
    }
}

File: frontend/src/types/invoice.ts
export interface Invoice {
  id: string
  patientId: string
  patientName: string
  procedureType: string
  totalAmount: number
  insuranceCoveredAmount: number
  patientAmount: number
  status: string
  issuedAt: string
  paidAt: string | null
}

export interface CreateInvoiceDto {
  patientId: string
  procedureType: string
}